paderbox/array/interval/core.py

```diff
@@ -466,33 +474,83 @@ def __setitem__(self, item, value):
         >>> ai
         ArrayInterval("0:10, 40:50", shape=(50,))
 
+        >>> ai = zeros(50)
+        >>> ai2 = zeros(10)
+        >>> ai2[5:10] = 1
+        >>> ai[10:20] = ai2
+        >>> ai
+        ArrayInterval("15:20", shape=(50,))
+        >>> ai2 = zeros(20)
+        >>> ai2[0:10] = 1
+        >>> ai2[15:20] = 1
+        >>> ai[10:30] = ai2
+        >>> ai
+        ArrayInterval("10:20, 25:30", shape=(50,))
+
+        >>> ai[40:60] = ai2
+        Traceback (most recent call last):
+          ...
+        ValueError: Could not broadcast input with length 20 into shape 10
+        >>> ai[-20:] = ai2
+        >>> ai
+        ArrayInterval("10:20, 25:40, 45:50", shape=(50,))
+
+        >>> ai = zeros(20)
+        >>> ai[:10] = ones(10)
+        >>> ai
+        ArrayInterval("0:10", shape=(20,))
+        >>> ai = ones(20)
+        >>> ai[:10] = zeros(10)
+        >>> ai
+        ArrayInterval("0:10", shape=(20,), inverse_mode=True)
+
         """
+        if not isinstance(item, slice):
+            raise NotImplementedError(
+                f'{self.__class__.__name__}.__setitem__ only supports slices '
+                f'for indexing, not {item}'
+            )
 
         start, stop = cy_parse_item(item, self.shape)
 
         if np.isscalar(value):
+            if value not in (0, 1):
+                # Numpy interprets values as boolean even if they are larger
+                # than 1, why doesn't the ArrayInterval do the same?
```

Comment:
Checking if the value is a Boolean (or 0, 1) is more verbose, and I don't know a case, where other values are useful. Usually they indicate a bug or a wrong assumption of the user.

Reply:
I updated the comment